tls/s2n_x509_validator.c

```diff
@@ -573,48 +578,32 @@ int s2n_disable_time_validation_ossl_verify_callback(int default_ossl_ret, X509_
     switch (err) {
         case X509_V_ERR_CERT_NOT_YET_VALID:
         case X509_V_ERR_CERT_HAS_EXPIRED:
-            return OSSL_VERIFY_CALLBACK_IGNORE_ERROR;
+            return OSSL_VERIFY_CALLBACK_OK;
         default:
             break;
     }
 
-    /* If CRL validation is enabled, setting the time validation verify callback will override the
-     * CRL verify callback. The CRL verify callback is manually triggered to work around this
-     * issue.
-     *
-     * The CRL verify callback ignores validation errors exclusively for CRL timestamp fields. So,
-     * if CRL validation isn't enabled, the CRL verify callback is a no-op.
-     */
-    return s2n_crl_ossl_verify_callback(default_ossl_ret, ctx);
+    return default_ossl_ret;
 }
 
-static S2N_RESULT s2n_x509_validator_disable_time_validation(struct s2n_connection *conn,
-        struct s2n_x509_validator *validator)
+int s2n_ossl_verify_callback(int default_ossl_ret, X509_STORE_CTX *ctx)
 {
-    RESULT_ENSURE_REF(conn);
-    RESULT_ENSURE_REF(conn->config);
-    RESULT_ENSURE_REF(validator);
-    RESULT_ENSURE_REF(validator->store_ctx);
-
-    /* Setting an X509_STORE verify callback is not recommended with AWS-LC:
-     * https://github.com/aws/aws-lc/blob/aa90e509f2e940916fbe9fdd469a4c90c51824f6/include/openssl/x509.h#L2980-L2990
-     *
-     * If the libcrypto supports the ability to disable time validation with an X509_VERIFY_PARAM
-     * NO_CHECK_TIME flag, this method is preferred.
-     *
-     * However, older versions of AWS-LC and OpenSSL 1.0.2 do not support this flag. In this case,
-     * an X509_STORE verify callback is used. This is acceptable in older versions of AWS-LC
-     * because the versions are fixed, and updates to AWS-LC will not break the callback
-     * implementation.
-     */
-    if (s2n_libcrypto_supports_flag_no_check_time()) {
-        RESULT_GUARD(s2n_x509_validator_set_no_check_time_flag(validator));
-    } else {
-        X509_STORE_CTX_set_verify_cb(validator->store_ctx,
-                s2n_disable_time_validation_ossl_verify_callback);
+    void *callback_config_paramter = X509_STORE_CTX_get_app_data(ctx);
+    if (callback_config_paramter == NULL) {
+        /* error, fail closed */
+        return OSSL_VERIFY_CALLBACK_ERROR;
     }
+    const struct s2n_ossl_verify_cb_config *callback_config = (struct s2n_ossl_verify_cb_config *) callback_config_paramter;
 
-    return S2N_RESULT_OK;
+    /* If any callback returns OK, then return OK */
```

Comment:
> but in general wouldn't we want to return OK only if all callbacks return OK?

Thinking about this more, I guess this is only really relevant when a callback can return an error, and none of them currently do that. So maybe handling this isn't needed yet.

Reply:
I'm not sure I understand the truth table, though I think it doesn't really matter at the moment because none of the callbacks can return errors. But would we ever want to return OK if a callback explicitly returns an error? Like this:
```
OSSL | CRL | TIME | SEC_POL
ERR  | ERR |  OK  |   OK    -> OK
```
Regardless of the OSSL return code, if the CRL callback found something wrong and errored why would we return OK?

Reply:
Ah ok I think I understand - the callbacks will return ERR even if they found nothing wrong if the OSSL return code was ERR.

Reply:
This needs a better comment. It's not understandable at a glance. Comments should explain "why" more than "what".